src/ring_theory/polynomial/gauss_lemma.lean

```diff
@@ -125,6 +129,37 @@ begin
   { left,
     apply is_unit_or_eq_zero_of_is_unit_integer_normalization_prim_part f h0.1 h },
 end
+
+lemma is_primitive.dvd_of_fraction_map_dvd_fraction_map {p q : polynomial R}
+  (hp : p.is_primitive) (hq : q.is_primitive) (h_dvd : p.map f.to_map ∣ q.map f.to_map) :
+  (p ∣ q) :=
+begin
+  rcases h_dvd with ⟨r, hr⟩,
+  obtain ⟨⟨s, s0⟩, hs⟩ := @integer_normalization_map_to_map _ _ _ _ _ f r,
+  rw [algebra.smul_def, ← C_eq_algebra_map, subtype.coe_mk] at hs,
+  have h : p ∣ q * C s,
+  { use (f.integer_normalization r),
+    apply map_injective _ f.injective,
+    rw [map_mul, map_mul, hs, hr, mul_assoc, mul_comm r],
+    simp },
+  rw [← hp.dvd_prim_part_iff_dvd, prim_part_mul, hq.prim_part_eq, dvd_iff_dvd_of_rel_right] at h,
+  { exact h },
+  { symmetry,
+    rcases is_unit_prim_part_C s with ⟨u, hu⟩,
+    use u,
+    simp [hu], },
+  iterate 2 { apply mul_ne_zero hq.ne_zero,
+    rw [ne.def, C_eq_zero],
+    contrapose! s0,
+    simp [s0, mem_non_zero_divisors_iff_ne_zero] }
+end
+
+lemma is_primitive.dvd_iff_fraction_map_dvd_fraction_map {p q : polynomial R}
```

Comment:
Name is swapped wrt statement

Reply:
I'm changing the statement, not the name.